consumer_group.go

```diff
@@ -682,6 +682,19 @@ func (s *consumerGroupSession) consume(topic string, partition int32) {
 		s.parent.handleError(err, topic, partition)
 	}
 
+	// update offset when auto commit
+	go func() {
+		for {
+			if !s.parent.config.Consumer.Offsets.AutoCommit.Enable {
+				continue
+			}
+			offsetFetched := claim.PartitionConsumer.OffsetFetched()
+			s.MarkOffset(topic, partition, offsetFetched, "")
+			time.Sleep(time.Second)
+		}
+	}()
+
```

Comment:
What are you doing?!  You mark all fetched messages consumed every second? And not respect to the closing event (`ctx.Done()` or `parent.closed`), which cause goroutine leak.

Reply:
Thanks a lot for you review.
I want to implement the auto-commit feature.  Yes, closing event should be respected.

Reply:
IIRC, the auto-commit feature already exists.

Reply:
See this thread, https://github.com/Shopify/sarama/issues/1570
The auto-commit feature won't work, and i have add analysis in the desc of this fix.